core/templates/pages/exploration-editor-page/services/learner-answer-details-data.service.ts

```diff
@@ -17,81 +17,101 @@
  * any learner answer info.
  */
 
-require('pages/exploration-editor-page/services/exploration-data.service.ts');
-require('domain/utilities/url-interpolation.service.ts');
-require('domain/statistics/LearnerAnswerInfoObjectFactory.ts');
-require('domain/statistics/LearnerAnswerDetailsObjectFactory.ts');
+import {ExplorationDataService} from
+  './exploration-data.service';
+import {UrlInterpolationService} from
+  '../../../domain/utilities/url-interpolation.service';
+import {LearnerAnswerInfo} from
+  '../../../domain/statistics/LearnerAnswerInfoObjectFactory';
+import {LearnerAnswerDetails} from
+  '../../../domain/statistics/LearnerAnswerDetailsObjectFactory';
+import {downgradeInjectable} from
+  '@angular/upgrade/static';  
+import {Injectable} from
+  '@angular/core';
+import {HttpClient} from
+  '@angular/common/http';
 
-angular.module('oppia').factory('LearnerAnswerDetailsDataService', [
-  '$http', '$q', 'ExplorationDataService', 'LearnerAnswerDetailsObjectFactory',
-  'LearnerAnswerInfoObjectFactory', 'UrlInterpolationService',
-  function(
-      $http, $q, ExplorationDataService, LearnerAnswerDetailsObjectFactory,
-      LearnerAnswerInfoObjectFactory, UrlInterpolationService) {
-    var _expId = ExplorationDataService.explorationId;
-    var _data = [];
-    var learnerAnswerInfoData = null;
-    var LEARNER_ANSWER_INFO_DATA_URL = (
+@Injectable( {
+  providedIn: 'root'
+})
+export class LearnerAnswerDetailsDataService {
+  
+  constructor(
+    private explorationDataService:ExplorationDataService,
+    private urlInterpolationService: UrlInterpolationService,
+    private http:HttpClient,
```

Comment:
Let keep this lexicographically sorted:
```suggestion
    private explorationDataService:ExplorationDataService,
    private http:HttpClient,
    private urlInterpolationService: UrlInterpolationService
```

Reply:
Done